packages/client/src/base.ts

```diff
@@ -72,8 +73,32 @@ export abstract class MessagingChannelBase extends Emitter<{
 
   private applyOptions() {
     const config = this.getAxiosConfig(this._options)
-    this.http = axios.create(config)
     this.adminHeader = this._options.adminKey?.length ? { 'x-bp-messaging-admin-key': this._options.adminKey } : {}
+
+    this.http = axios.create(config)
+    this.http.interceptors.response.use(
+      (e) => {
+        this.saveCookie(e.headers['set-cookie'])
+        return e
+      },
+      (e) => {
+        this.saveCookie(e?.response?.headers?.['set-cookie'])
+        return Promise.reject(e)
+      }
+    )
+  }
+
+  private saveCookie(cookieHeader: string[] | undefined) {
+    if (!cookieHeader) {
+      return
+    }
+
+    for (const strCookie of cookieHeader) {
+      const resCookie = cookie.parse(strCookie)
+      if (resCookie.bp_messaging) {
```

Comment:
The name of the `cookie` should be extracted in a configuration setting this way it is going to be easier to change it without having to rebuild a new image of the runtime.

```
const config = {
   [...]
   messagingSessionCookie: process.env.MESSAGING_SESSION_COOKIE ?? 'bp_messaging'
}
```